third_party/FMI/CMakeLists.txt

```diff
@@ -22,18 +21,14 @@ SET(SRC
   fmiModelTypes.h
 )
 
-ADD_LIBRARY( epfmiimport STATIC ${SRC} )
+add_library(epfmiimport STATIC ${SRC})
 
-IF (UNIX)
-  ADD_DEFINITIONS("-fPIC")
-ENDIF()
+if(UNIX)
+  set_property(TARGET epfmiimport PROPERTY POSITION_INDEPENDENT_CODE ON)
+endif()
 
-TARGET_LINK_LIBRARIES( epfmiimport epexpat miniziplib )
+target_link_libraries(epfmiimport PRIVATE epexpat miniziplib)
```

Comment:
Couldn't mix the old signature with the new that has PUBLIC/PRIVATE/INTERFACE, and probably why it was the only target linking to turn_off_warnings that wasn't linking with PRIVATE